jaxrs-clients/src/main/java/com/palantir/remoting1/jaxrs/FeignJaxRsClientBuilder.java

```diff
@@ -80,11 +81,12 @@
     FeignJaxRsClientBuilder(ClientConfig config, BackoffStrategy backoffStrategy) {
         this.backoffStrategy = backoffStrategy;
         this.config = config;
+        Preconditions.checkArgument(config.maxNumRetries() == 0,
```

Comment:
If we're making this change, is there any reason to even have `backoffStrategy` as an ivar? Seems like we can just define it as a static final constant with the value `NeverRetryingBackoffStrategy.INSTANCE`.